store/snap_remote_repo.go

```diff
@@ -278,7 +293,17 @@ func (s *SnapUbuntuStoreRepository) Snap(name, channel string) (*snap.Info, erro
 		return nil, err
 	}
 
-	return infoFromRemote(detailsData), nil
+	suggestedCurrency := getSuggestedCurrency(&resp.Header)
+
+	return infoFromRemote(detailsData, suggestedCurrency), nil
+}
+
+func getSuggestedCurrency(h *http.Header) string {
+	s := h.Get("X-Suggested-Currency")
+	if s == "" {
+		s = "USD"
```

Comment:
It's what the current store-front on the phone does (unity-scope-click). I don't know what other behaviour to apply.